### PR TITLE
Addition of New OpticalFlow.msg

### PR DESCRIPTION
**Pull request by Vladislavert**
This PR proposes the addition of a new OpticalFlow.msg file to enhance the [optical flow](https://mavlink.io/en/messages/common.html#OPTICAL_FLOW) support in MAVROS

### CODE REVIEW

File: mavros_extras/src/plugins/optical_flow.cpp [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+    auto int_flow_comp_m_xy = ftf::transform_frame_aircraft_baselink(
+      Eigen::Vector3d(
+        flow.flow_comp_m_x,
+        flow.flow_comp_m_y,
+        0.0));
+    auto int_flow_rate_xy = ftf::transform_frame_aircraft_baselink(
+      Eigen::Vector3d(
+        flow.flow_rate_x,
+        flow.flow_rate_y,
+        0.0));
```
Comment (vooon) [inline mavros_extras/src/plugins/optical_flow.cpp:131]: After that conversion i'm not sure that we can just keep XY. Maybe better Vector3?

File: mavros_msgs/msg/OpticalFlow.msg [hunk 2]
```diff
@@ -0,0 +1,9 @@
+# OPTICAL_FLOW message data
+
+std_msgs/Header header
+
+int16[2] flow
+float32[2] flow_comp_m
+uint8 quality
+float32 ground_distance
+float32[2] flow_rate
```
Comment (vooon) [inline mavros_msgs/msg/OpticalFlow.msg:9]: Usually ROS recommend to use full Vector3 and just don't set Z.

Comment (vooon) [reply mavros_msgs/msg/OpticalFlow.msg:9]: Well, arrays looks worse than _x/_y.
But, why not to try this:
```
geometry_msgs/Vector3 flow
Vector3 flow_comp_m
Vector3 flow_rate
```

**Review by vooon — CHANGES_REQUESTED** [review]

File: mavros_msgs/msg/OpticalFlow.msg (hunk 2 above)
Comment (Vladislavert) [reply mavros_msgs/msg/OpticalFlow.msg]: flow in mavlink is int16_t, convert to float?

File: mavros_extras/src/plugins/optical_flow.cpp (hunk 1 above)
Comment (Vladislavert) [reply mavros_extras/src/plugins/optical_flow.cpp:133]: The z component of the vector is not used in the optical flow algorithm. The function `ftf::transform_frame_aircraft_baselink`, which multiplies the vector by `quaternion_from_rpy(M_PI, 0.0, 0.0)`, it becomes clear that the z component does not contribute to the algorithm. Consequently, it is recommended to preserve only the XY components of the vector and explicitly set the z component to 0.0 during the conversion process. This approach aligns with the requirements of the algorithm and ensures proper functioning.

File: mavros_msgs/msg/OpticalFlow.msg (hunk 2 above)
Comment (vooon) [reply mavros_msgs/msg/OpticalFlow.msg]: I think yes, easier to use existing and well established type. We can define variety of Vector2, I16Vector2 and so, but for other ros users that anyway would need more and more conversions.

**Review by vooon — APPROVED** [review]
LGTM, Thanks!

**Pull request merged**